cellxgene_schema_cli/cellxgene_schema/validate.py

```diff
@@ -182,6 +182,39 @@ def _validate_curie_ontology(self, term_id: str, column_name: str, allowed_ontol
                 f"'{term_id}' in '{column_name}' is not a valid ontology term id of '{', '.join(allowed_ontologies)}'."
             )
 
+    def _validate_curie_str(self, term_str: str, column_name, curie_constraints: dict):
+        """
+        Validate a curie str based on some constraints. If there are any errors, it adds them to self.errors
+
+        :param str term_str: the curie term str to validate
+        :param str column_name: Name of the column in the dataframe
+        :param dict curie_constraints: constraints for the curie term to be validated,
+        this part of the schema definition
+
+        :rtype None
+        """
+        # If there are exceptions and this is one then skip to end
+        if "exceptions" in curie_constraints and term_str in curie_constraints["exceptions"]:
+            return
+
+        # If NA is found in allowed ontologies, it means only exceptions should be found. If no exceptions were found
+        # then return error
+        if curie_constraints["ontologies"] == ["NA"]:
```

Comment:
moved from '_validate_curie' to keep exception-logic together in top-level function

Reply:
I'm a little confused wrt `"NA"` here. I do not see reference to `"NA"` anywhere in the schema, but I do see reference to `"na"`. I just want to verify that you are intentionally using `"NA"` in some cases and `"na"` in others, esp since `"na"` is specified in the reqs for this issue, but it is less common to see it as lowercase.

Reply:
Ah. I see this has been specified differentially in the schema. Nvm.

Reply:
it is intentional, but not introduced in this PR. This block is being moved from a different function to factor out some shared checks. We can make it consistent, though offhand I'm not sure what the original reason for making them different was.